core/core-impl/src/main/java/ai/timefold/solver/core/impl/localsearch/DefaultLocalSearchPhaseFactory.java

```diff
@@ -191,18 +198,112 @@ private UnionMoveSelectorConfig determineDefaultMoveSelectorConfig(HeuristicConf
+                                                    .withNearbySelectionConfig(new NearbySelectionConfig()
+                                                            .withOriginEntitySelectorConfig(new EntitySelectorConfig()
+                                                                    .withMimicSelectorRef("tailChainSwapMoveSelector"))
+                                                            .withNearbyDistanceMeterClass(
+                                                                    configPolicy.getNearbyDistanceMeterClass()))));
+                } else {
+                    return new UnionMoveSelectorConfig()
+                            .withMoveSelectors(new ChangeMoveSelectorConfig(),
+                                    new SwapMoveSelectorConfig(),
+                                    new ChangeMoveSelectorConfig()
+                                            .withEntitySelectorConfig(new EntitySelectorConfig().withId("changeMoveSelector"))
+                                            .withValueSelectorConfig(new ValueSelectorConfig()
+                                                    .withNearbySelectionConfig(new NearbySelectionConfig()
+                                                            .withOriginEntitySelectorConfig(new EntitySelectorConfig()
+                                                                    .withMimicSelectorRef("changeMoveSelector"))
+                                                            .withNearbyDistanceMeterClass(
+                                                                    configPolicy.getNearbyDistanceMeterClass()))),
+                                    new SwapMoveSelectorConfig()
+                                            .withEntitySelectorConfig(new EntitySelectorConfig()
+                                                    .withId("swapMoveSelector"))
+                                            .withSecondaryEntitySelectorConfig(new EntitySelectorConfig()
+                                                    .withNearbySelectionConfig(new NearbySelectionConfig()
+                                                            .withOriginEntitySelectorConfig(new EntitySelectorConfig()
+                                                                    .withMimicSelectorRef("swapMoveSelector"))
+                                                            .withNearbyDistanceMeterClass(
+                                                                    configPolicy.getNearbyDistanceMeterClass()))));
+                }
+            }
         } else {
             /*
              * We have a mix of basic and list variables.
              * The "old" move selector configs handle this situation correctly but they complain of it being deprecated.
              *
+             * Combining essential variables with list variables in a single entity is not supported. Therefore,
```

Comment:
How can this even happen? List and basic vars are mutually incompatible - so shouldn't this `else` branch fail fast instead? Or is there a place where this actually works?